rest-service/manager_rest/server.py

```diff
@@ -151,6 +170,55 @@ def reset_state(configuration=None):
     app = setup_app()
 
 
+def validate_security_config(cfy_config):
+    # TODO raise better exceptions
+    if not hasattr(cfy_config, 'securest_secret_key') or \
+            cfy_config.securest_secret_key is None:
+        raise Exception('securest_secret_key not set')
+    if not hasattr(cfy_config, 'securest_authentication_methods') or \
+            cfy_config.securest_authentication_methods is None:
+        raise Exception('securest_authentication_methods not set')
+    if not hasattr(cfy_config, 'securest_userstore_driver') or \
+            cfy_config.securest_userstore_driver is None:
+        raise Exception('securest_userstore_driver not set')
+    if not hasattr(cfy_config, 'securest_userstore_identifier_attribute') or \
+            cfy_config.securest_userstore_identifier_attribute is None:
+        raise Exception('securest_userstore_identifier_attribute not set')
+
+
+def init_secure_app(app):
+    cfy_config = config.instance()
+    validate_security_config(cfy_config)
+
+    app.config[SECUREST_SECRET_KEY] = cfy_config.securest_secret_key
+    app.config[SECUREST_USERSTORE_DRIVER] = \
+        cfy_config.securest_userstore_driver
+    app.config[SECUREST_USERSTORE_IDENTIFIER_ATTRIBUTE] = \
+        cfy_config.securest_userstore_identifier_attribute
+
+    secure_app = SecuREST(app)
+
+    register_authentication_methods(secure_app,
+                                    cfy_config.securest_authentication_methods)
+
+    @secure_app.unauthorized_user_handler
+    def unauthorized_user_handler():
+        print '**** GOTCHA!'
+        raise Exception(401)
+
+
+def register_authentication_methods(secure_app, authentication_providers):
+    # Note: the order of registration is important here
+    for auth_method_path in authentication_providers:
+        try:
+            auth_provider = util.get_class_instance(auth_method_path)
+            secure_app.authentication_provider(auth_provider)
+        except Exception as e:
+            print('Failed to register authentication method: ',
+                  authentication_providers, e)
+            raise e
```

Comment:
when re-raising, don't do raise e because you will lose the traceback, simply raise